PG.StarWarsGame.Components.Localisation/PG.StarWarsGame.Components.Localisation/Languages/AlamoLanguageDefinitionBase.cs

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Alamo Engine Tools and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+using System;
+using System.Globalization;
+using FluentValidation;
+using FluentValidation.Results;
+using PG.Commons.Validation;
+
+namespace PG.StarWarsGame.Components.Localisation.Languages;
+
+/// <summary>
+///     Base definition for an <see cref="IAlamoLanguageDefinition" /> implementation.
+/// </summary>
+public abstract class AlamoLanguageDefinitionBase : IAlamoLanguageDefinition, IValidatable
+{
+    /// <summary>
+    ///     A string that is being used to identify the language of the *.DAT file, e.g. a language identifier
+    ///     "english" would produce the file "mastertextfile_english.dat"
+    /// </summary>
+    protected abstract string ConfiguredLanguageIdentifier { get; }
+
+    /// <summary>
+    ///     The .NET Culture that best describes the language. This culture can be used for spell checking,
+    ///     auto-translation between languages, etc.
+    /// </summary>
+    protected abstract CultureInfo ConfiguredCulture { get; }
+
+    /// <inheritdoc />
+    public string LanguageIdentifier => ConfiguredLanguageIdentifier;
+
+    /// <inheritdoc />
+    public CultureInfo Culture => ConfiguredCulture;
+
+    /// <summary>
+    ///     Determines whether the specified object is equal to the current object.
+    /// </summary>
+    /// <param name="other"></param>
+    /// <returns></returns>
+    protected virtual bool EqualsInternal(IAlamoLanguageDefinition other)
+    {
+        return GetHashCodeInternal().CompareTo(other.GetHashCode()) == 0;
```

Comment:
Is it usual to have the equals method be dependent on the hashcode method?

Reply:
Not usual but not exactly unusual either. Usually it would be something like `a.equals(b) ⇒ a.hashCode() == b.hashCode();` But ` a.hashCode() == b.hashCode ⇏ a.equals(b);` In this case I actively want to enforce that the hash code and equality are the same.